bigml/ensemble.py

```diff
@@ -280,8 +301,27 @@ def all_model_fields(self):
         """
 
         fields = {}
-        for model_id in self.model_ids:
-            local_model = Model(model_id, self.api)
+        models = []
+        objective_id = None
+        no_objective_id = False
+        if isinstance(self.models_splits[0][0], Model):
+            for split in self.models_splits:
+                models.extend(split)
+        else:
+            models = self.model_ids
+        for model_id in models:
+            if isinstance(model_id, Model):
+                local_model = model_id
+            else:
+                local_model = Model(model_id, self.api)
             gc.collect()
             fields.update(local_model.fields)
-        return fields
+            if (objective_id is not None and
+                    objective_id != local_model.objective_id):
+                # the models' objective field have different ids, no global id
+                no_objective_id = True
```

Comment:
why don't you just assign here `objective_id=None` and do without the auxiliary variable `no_objective_id`? i think it'd be easier to follow...